### PR TITLE
Fix NOHOST flag on IPNetwork copy constructor

### PR DESCRIPTION
**Pull request by claytonsingh**
Add tests for NOHOST flag
Fixes https://github.com/netaddr/netaddr/issues/374

### CODE REVIEW

File: netaddr/tests/ip/test_ip_v4.py [hunk 1]
```diff
@@ -487,3 +487,8 @@ def test_spanning_cidr_handles_strings():
     ]
     assert spanning_cidr(addresses) == IPNetwork('10.0.0.0/29')
     assert spanning_cidr(reversed(addresses)) == IPNetwork('10.0.0.0/29')
+
+
+def test_ipnetwork_nohost():
+    assert IPNetwork('192.168.0.1/24', flags=NOHOST).ip == IPAddress('192.168.0.0')
```
Comment (jstasiak) [inline netaddr/tests/ip/test_ip_v4.py:493]: Can you please remove this line? Yes, it's a test worth having but it's a test out of scope of this PR (which is about fixing the copy constructor).

Comment (claytonsingh) [reply netaddr/tests/ip/test_ip_v4.py]: I have removed the out of scope test

**Comment by jstasiak** [comment]
Thank you

**Comment by jstasiak** [comment]
Released in netaddr 1.3.0.

**Pull request merged**